src/scripts/youtube-info.coffee

```diff
@@ -33,28 +36,36 @@ module.exports = (robot) ->
     showInfo msg, video_hash
 
 showInfo = (msg, video_hash) ->
-  msg.http("http://gdata.youtube.com/feeds/api/videos/#{video_hash}")
+  msg.http("https://www.googleapis.com/youtube/v3/videos?part=snippet,contentDetails,statistics&id=#{video_hash}&key=#{apiKey}")
     .query({
       alt: 'json'
     }).get() (err, res, body) ->
       if res.statusCode is 200
-        data = JSON.parse(body)
-        entry = data.entry
-        r = entry.gd$rating
-        thumbs_up = Math.round(((r.average-r.min)/(r.max-r.min))*r.numRaters)
-        thumbs_down = r.numRaters - thumbs_up
-        msg.send "YouTube: #{entry.title.$t} (#{formatTime(entry.media$group.yt$duration.seconds)}, #{humanizeNumber(entry.yt$statistics.viewCount)} views, #{humanizeNumber(thumbs_up)} thumbs up, #{humanizeNumber(thumbs_down)} thumbs down)"
+        data = JSON.parse(body).items[0]
+        title = data.snippet.title
+        views = humanizeNumber(data.statistics.viewCount)
+        thumbs_up = humanizeNumber(data.statistics.likeCount)
+        thumbs_down = humanizeNumber(data.statistics.dislikeCount)
+        time = formatTime(data.contentDetails.duration)
+        msg.send "YouTube: #{title} (#{time}, #{views} views, #{thumbs_up} thumbs up, #{thumbs_down} thumbs down)"
       else
         msg.send "YouTube: error: #{video_hash} returned #{res.statusCode}: #{body}"
 
-formatTime = (seconds) ->
-  min = Math.floor(seconds / 60)
-  sec = seconds % 60
+formatTime = (duration) ->
+  # I'm not good enough at regex groups to make this any better
```

Comment:
If the input is: `PT10H5M16S` ([ISO 8601 duration](https://en.wikipedia.org/wiki/ISO_8601#Durations)) where any part could be missing, then you can use:

```coffeescript
hmsPattern = /^PT((\d+)H)?((\d+)M)?((\d+)S)?/
hmsMatch = duration.match(hmsPattern)
hours = parseInt hmsMatch[2] or 0
min = parseInt hmsMatch[4] or 0
sec = parseInt hmsMatch[6] or 0
```

Running that in the REPL returns:

```
coffee> duration = 'PT10H16S'
'PT10H16S'
coffee> hmsPattern = /^PT((\d+)H)?((\d+)M)?((\d+)S)?/
/^PT((\d+)H)?((\d+)M)?((\d+)S)?/
coffee> hmsMatch = duration.match(hmsPattern)
[ 'PT10H16S',
  '10H',
  '10',
  undefined,
  undefined,
  '16S',
  '16',
  index: 0,
  input: 'PT10H16S' ]
coffee> hours = parseInt hmsMatch[2] or 0
10
coffee> min = parseInt hmsMatch[4] or 0
0
coffee> sec = parseInt hmsMatch[6] or 0
16
```

Though, if your input format is as described, this is a lazy version:

```coffeescript
formatTime = (duration) ->
  match = duration.match(/^PT(.*)$/)?[1] or ''
  match.toLowerCase()
```